From d21b865e64827de95a27cf27241225a9060a5c7a Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Mon, 11 Jan 2021 18:33:12 +0200
Subject: [PATCH] [SW-20719] gaudi: export device id function

The preboot_get_vendor_id_device_id() funciton is useful for
differentiating the different Gaudi SoC IDs, so the actual code behavior
can be altered accordingly to the ID.
Export the function.

Change-Id: I12a651d486010350a0b6473ead01e581b887f3f1
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/common.c b/gaudi/common.c
index 036650e..7faf702 100644
--- a/gaudi/common.c
+++ b/gaudi/common.c
@@ -109,7 +109,7 @@
 
 #define FUSE_DEV_ID_WORD_OFFSET	1
 
-static u32 preboot_get_vendor_id_device_id(void)
+u32 preboot_get_vendor_id_device_id(void)
 {
 	u32 pci_dev_id = PCI_IDS_GAUDI;
 	char devid[5];
diff --git a/gaudi/common.h b/gaudi/common.h
index 0b2f946..cc43c31 100644
--- a/gaudi/common.h
+++ b/gaudi/common.h
@@ -42,6 +42,7 @@
 
 #ifndef __ASSEMBLY__
 #include <spi.h>
+#include <types.h>
 
 /* Let this file be included in any order and override the PROT_MASK */
 #ifdef PROT_MASK
@@ -50,5 +51,7 @@
 /* Only BP2 == 0x4 - means 8 blocks - up to 512KB */
 #define PROT_MASK			(READ_STATUS_BP2 | READ_STATUS_SRWD)
 
+u32 preboot_get_vendor_id_device_id(void);
+
 #endif /* __ASSEMBLY__ */
 #endif /* COMMON_GAUDI_H_ */
